src/Microsoft.Extensions.Primitives/ChangeToken.cs

```diff
@@ -36,7 +36,7 @@ public static IDisposable OnChange(Func<IChangeToken> changeTokenProducer, Actio
                 // If the token changes after we take the token, then we'll process the update immediately upon
                 // registering the callback.
                 var t = changeTokenProducer();
-                changeTokenConsumer();
+                try { changeTokenConsumer(); } catch { } // Uncaught exceptions are really bad here
```

Comment:
Rethrowing is better if there was any way for them to see the exception. Unfortunately there is nothing they can do here unless we invent something akin to our load error context which seems like overkill for something this low level. They can always put that logic in the consumer code like we are doing.